### PR TITLE
Update block tests to Sepolia and add more tests

### LINKED ISSUE #544: goerli depreceation
Any tests and examples that currently use Goerli should use another test network, like Sepolia. 

1. Enter the rpc directory
2. rename .env.template to .env.testnet, and enter the Sepolia endpoint
3. go test ./... -env testnet

You will see a bunch of tests that currently depend on Goerli, but should be updated to depend on Sepolia instead.

Free RPC endpoints https://data.voyager.online/

### PR DESCRIPTION
**Pull request by AryanGodara**
Partially resolves #544

### CODE REVIEW

**Comment by AryanGodara** [comment]
Most changes in this PR were already merged in a separate PR by @thiagodeev 
But, the newly added transaction types are missing.
I'll update and rebase this branch, to include those new transaction types

**Comment by thiagodeev** [comment]
Hey @AryanGodara , you already updated the transaction types in the PR [#557](https://github.com/NethermindEth/starknet.go/pull/557/files#diff-9d9c8e1eb95788fa7f5a85ae73e9187966d5364150405387969bf632b1c4f40f). If it's only that, you can close this PR since your modifications is already merged on the main

File: rpc/block_test.go [hunk 1]
```diff
@@ -677,130 +952,31 @@ func TestCaptureUnsupportedBlockTxn(t *testing.T) {
...
-				if !okv0 && !okv1 && !okl1 && !okdec0 && !okdec1 && !okdec2 && !okdep && !okdepac {
+				if !okv0 && !okv1 && !okv3 && !okl1 && !okdec0 && !okdec1 && !okdec2 && !okdec3 && !okdep && !okdepac {
 					t.Fatalf("New Type Detected %T at Block(%d)/Txn(%d)", v, i, k)
 				}
 			}
 		}
 	}
 }
 
-// TestBlockWithTxsAndInvokeTXNV1 is a test function that tests the behavior of the BlockWithTxsAndInvokeTXNV1 function.
```
Comment (thiagodeev) [inline rpc/block_test.go:696]: Hey @AryanGodara , why did you remove the already updated test `TestBlockWithTxsAndInvokeTXNV1` to create another one?

Comment (AryanGodara) [reply rpc/block_test.go:696]: Check now, it's only shifted not changed. (so that v0,v1 and v3 are together)

File: rpc/block_test.go [hunk 2]
```diff
@@ -353,6 +351,279 @@ func TestBlockWithTxsAndInvokeTXNV0(t *testing.T) {
...
+// It then initializes a fullBlockSepolia52767 variable with a Block struct and invokes the BlockWithTxs function with different test scenarios.
+// The function compares the expected error with the actual error and checks if the BlockWithTxs function returns the correct block data.
+// It also verifies the block hash, the number of transactions in the block, and the details of a specific transaction.
+//
+// Parameters:
+// - t: The t testing object
+// Returns:
+//
+//	none
+func TestBlockWithTxsAndInvokeTXNV3(t *testing.T) {
```
Comment (thiagodeev) [inline rpc/block_test.go:456]: The `TestBlockWithTxsAndInvokeTXNV3` must be based on the existing [TestBlockWithTxsAndInvokeTXNV1 ](https://github.com/NethermindEth/starknet.go/blob/main/rpc/block_test.go#L703)test because it was already updated with the requirements specified in these comments https://github.com/NethermindEth/starknet.go/pull/566#discussion_r1598451390

File: rpc/block_test.go (hunk 1 above)
Comment (thiagodeev) [reply rpc/block_test.go:696]: Great. Your new v1 was different from the v1 already existing in the repo, but in your last commit you fixed it.

File: rpc/block_test.go [hunk 3]
```diff
@@ -554,47 +557,22 @@ func TestBlockWithTxsAndInvokeTXNV3(t *testing.T) {
...
+		invokeV3Block, ok := blockWithTxs.Transactions[test.LookupTxnPositionInOriginal].(BlockInvokeTxnV3)
+		require.True(t, ok, "Expected invoke v3 transaction.")
 
-			require.Equal(t, invokeV3Want.TransactionHash, invokeV3Block.TransactionHash, "expected equal TransactionHash")
-			require.Equal(t, invokeV3Want.InvokeTxnV3.NonceDataMode, invokeV3Block.InvokeTxnV3.NonceDataMode, "expected equal nonceDataMode")
-			require.Equal(t, invokeV3Want.InvokeTxnV3.FeeMode, invokeV3Block.InvokeTxnV3.FeeMode, "expected equal feeMode")
-		}
+		require.Equal(t, invokeV3Want.TransactionHash.String(), invokeV3Block.TransactionHash.String(), "Expected equal TransactionHash.")
+		// require.Equal(t, invokeV3Want.InvokeTxnV3.Nonce, invokeV3Block.InvokeTxnV3.Nonce, "Expected equal nonce.")
+		// require.Equal(t, invokeV3Want.InvokeTxnV3.Calldata[1].String(), invokeV3Block.InvokeTxnV3.Calldata[1].String(), "Expected equal calldatas.")
```
Comment (thiagodeev) [inline rpc/block_test.go]: Why those comments? Either uncomment or remove them

File: rpc/block_test.go (hunk 2 above)
Comment (AryanGodara) [reply rpc/block_test.go:456]: I've pushed the updated test, which is now passing locally :D

File: rpc/block_test.go (hunk 3 above)
Comment (AryanGodara) [reply rpc/block_test.go]: This was an issue with wrong nonce in testcase, I forgot this by mistake, and have pushed the updated code ✅

File: rpc/block_test.go [hunk 4]
```diff
@@ -616,6 +858,10 @@ func TestBlockTransactionCount(t *testing.T) {
 				BlockID:       WithBlockNumber(30000),
 				ExpectedCount: 4,
 			},
+			{
+				BlockID:       WithBlockNumber(52959),
```
Comment (rianhughes) [inline rpc/block_test.go:833]: Would you mind updating the assertions used in the test here? We should be able to clean it up quite a bit by using require.Equal() / require.NoError() etc instead of all the "if err, t.Fatal()" code. Thanks

Comment (AryanGodara) [reply rpc/block_test.go:833]: I'll get on it🫡

File: rpc/block_test.go [hunk 5]
```diff
@@ -677,130 +923,31 @@ func TestCaptureUnsupportedBlockTxn(t *testing.T) {
...
-		invokeV1Block, ok := blockWithTxs.Transactions[test.LookupTxnPositionInOriginal].(BlockInvokeTxnV1)
-		require.True(ok, "Expected invoke v1 transaction.")
-
-		require.Equal(invokeV1Want.TransactionHash.String(), invokeV1Block.TransactionHash.String(), "Expected equal TransactionHash.")
-		require.Equal(invokeV1Want.InvokeTxnV1.MaxFee.String(), invokeV1Block.InvokeTxnV1.MaxFee.String(), "Expected equal maxfee.")
-		require.Equal(invokeV1Want.InvokeTxnV1.Calldata[1].String(), invokeV1Block.InvokeTxnV1.Calldata[1].String(), "Expected equal calldatas.")
-	}
-}
-
 // TestStateUpdate is a test function for the StateUpdate method.
```
Comment (rianhughes) [inline rpc/block_test.go:910]: The mock test data here is from the mock network. Can we update it to use sepolia data please?

**Review by rianhughes — APPROVED** [review]
